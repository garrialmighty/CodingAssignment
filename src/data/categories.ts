import {Category} from 'src/redux/reducer/types';

export const DefaultCategories: Category[] = [
  {
    name: 'Cities',
    items: [
      {
        description: 'Capital of Singapore',
        answer: 'SINGAPORE',
        points: 10,
      },
      {
        description: 'Capital of France',
        answer: 'PARIS',
        points: 6,
      },
    ],
  },
  {
    name: 'Food',
    items: [
      {
        description: 'A spicy noodle dish popular in Southeast Asia.',
        answer: 'LAKSA',
        points: 6,
      },
    ],
  },
  {
    name: 'Animals',
    items: [
      {
        description: 'Insects that have large, often, brightly coloured wings.',
        answer: 'BUTTERFLY',
        points: 10,
      },
    ],
  },
  {
    name: 'Movies',
    items: [
      {
        description:
          '2012 American superhero film based on the Marvel comics of the same name.',
        answer: 'AVENGERS',
        points: 9,
      },
    ],
  },
];
