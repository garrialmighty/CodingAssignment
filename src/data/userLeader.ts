import {Leader} from 'src/redux/reducer/types';

export const UserLeader: Leader = {
  points: 0,
  position: 1,
  name: 'Current User',
};
