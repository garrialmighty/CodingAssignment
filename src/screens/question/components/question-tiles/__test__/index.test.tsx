import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';

import QuestionTiles from '../';
import {DefaultCategories} from 'src/data/categories';

const mockAnswer = [...DefaultCategories[0].items[0].answer];

describe('Component: QuestionTiles', () => {
  it('should render properly', () => {
    const snapshot = render(<QuestionTiles letters={mockAnswer} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('can handle press events', () => {
    const mockPress = jest.fn();
    const {getByTestId} = render(
      <QuestionTiles letters={mockAnswer} onPress={mockPress} />,
    );
    fireEvent.press(getByTestId(`category-button-${mockAnswer[0]}`));
    expect(mockPress).toHaveBeenCalledTimes(1);
  });
});
