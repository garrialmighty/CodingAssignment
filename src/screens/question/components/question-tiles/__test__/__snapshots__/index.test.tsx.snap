// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: QuestionTiles should render properly 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "rowGap": 10,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-S"
  >
    <Text>
      S
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-I"
  >
    <Text>
      I
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-N"
  >
    <Text>
      N
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-G"
  >
    <Text>
      G
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-A"
  >
    <Text>
      A
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-P"
  >
    <Text>
      P
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-O"
  >
    <Text>
      O
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-R"
  >
    <Text>
      R
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 45,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "width": 45,
      }
    }
    testID="category-button-E"
  >
    <Text>
      E
    </Text>
  </View>
</View>
`;
